### PR TITLE
Standardize error_detail naming

### PR DESCRIPTION
**Pull request by delputnam**
This PR standardizes the event error data and adds methods to access the additional event data.

Also updates the docs and tests with these changes.

Testing:
Inspect the code, make sure it matches the data that will be returned by the server.
Run the unit tests: `composer update; ./vendor/bin/phpunit -c ./test/phpunit.xml`

### CODE REVIEW

File: test/lib/domain-services-client-test-case.php [hunk 1]
```diff
@@ -50,6 +50,28 @@ public function assertIsValidResponse( array $expected_data, Response\Response_I
...
+		$this->assertEquals( $expected_event_data['event_class'], $event->get_event_class() );
+		$this->assertEquals( $expected_event_data['event_subclass'], $event->get_event_subclass() );
+		$this->assertEquals( $expected_event_data['object_type'], $event->get_object_type() );
+		$this->assertEquals( $expected_event_data['object_id'], $event->get_object_id() );
+		$this->assertEquals( $expected_event_data['event_date'], Helper\Date_Time::format( $event->get_event_date() ) );
+		$this->assertEquals( $expected_event_data['acknowledged_date'], null === $event->get_acknowledged_date() ? null : Helper\Date_Time::format( $event->get_acknowledged_date() ) );
+		$this->assertEquals( $expected_event_data['event_data']['status'], $event->get_event_status() );
+		$this->assertEquals( $expected_event_data['event_data']['status_description'], $event->get_event_status_description() );
+		$this->assertEquals( $expected_event_data['event_data']['client_txn_id'], $event->get_event_client_txn_id() );
+		$this->assertEquals( $expected_event_data['event_data']['server_txn_id'], $event->get_event_server_txn_id() );
```
Comment (leonardost) [inline test/lib/domain-services-client-test-case.php]: When running tests, these four lines reported "undefined array key" errors. Maybe we could add null coalescing to them? Or I guess the correct way would be to add these properties to all test cases, but I think adding `?? null` would be fine 🤔

**Review by leonardost — APPROVED** [review]
LGTM! Code changes look good, spot checked docs and they seem ok too. Left just one comment regarding unit tests.

File: test/lib/domain-services-client-test-case.php (hunk 1 above)
Comment (delputnam) [reply test/lib/domain-services-client-test-case.php]: Good catch!  There are some events that don't have these properties.  I've added the null coalesce.
